app/views/layouts/application.html.erb

```diff
@@ -21,6 +21,47 @@
         <%= @hero_html.html_safe %>
       </div>
     <% end %>
+    
+    <meta name="page-cached-at" content="<%= Time.current.to_i %>">
+    <meta name="proper-stylesheet-crayons" content="<%= stylesheet_path "crayons" %>">
+    <meta name="proper-stylesheet-views" content="<%= stylesheet_path "views" %>">
+    <meta name="proper-stylesheet-minimal" content="<%= stylesheet_path "minimal" %>">
+    <meta name="proper-js-base" content="<%= javascript_url "base" %>">
+
+    <script defer>
+
+      // This area is designed to reconcile issues of mismatched caches stylesheets.
+      // Currently it handles a scenario where the head assets are out of date and need to be updated.
+      // There is a legitimate reverse scenario we do not yet handle where we *should* use the "out-of-date"
+      // content on the page, but since we may have issues persisting old stylesheets etc.
+      // this only handles the most troubling issues of pages that load with no styles.
+
+      var headCacheCheck = document.querySelector("meta[name='head-cached-at']");
+      var pageCacheCheck = document.querySelector("meta[name='page-cached-at']");
+      var headCrayonsPath = document.getElementById("main-crayons-stylesheet")
+      var headViewsPath = document.getElementById("main-views-stylesheet")
+      var headMinimalPath = document.getElementById("main-minimal-stylesheet")
+      var pageCrayonsPath = document.querySelector("meta[name='proper-stylesheet-crayons']");
+      var pageViewsPath = document.querySelector("meta[name='proper-stylesheet-views']");
+      var pageMinimalPath = document.querySelector("meta[name='proper-stylesheet-minimal']");
+
+      var headBaseJSPath = document.getElementById("main-base-javascript")
+      var pageBaseJSPath = document.querySelector("proper-js-base")
```

Comment:
I'm a little worried about hardcoding all of this (and then having to remember to change it whenever crayons change). Also thinking about the eventual "extract out crayons" path here, too. 🤔

Reply:
Because this is dealing with changing conditions based on caching, I'm not sure what a good way to test this might be. I'm all for answers here!

Reply:
@vaidehijoshi I added a regression test, as well as moved this code to a partial to clean up the main view.

It still doesn't test all the logic, but ensures if we change something that shouldn't be changed, it will be caught in the test.